src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -1,102 +1,311 @@
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.ItemType;
 import org.openstreetmap.atlas.geography.atlas.items.Point;
-import org.openstreetmap.atlas.geography.atlas.items.Relation;
-import org.openstreetmap.atlas.geography.atlas.raw.slicing.changeset.RelationChangeSet;
-import org.openstreetmap.atlas.geography.atlas.raw.slicing.changeset.SimpleChangeSet;
 import org.openstreetmap.atlas.geography.boundary.CountryBoundaryMap;
+import org.openstreetmap.atlas.geography.sharding.Shard;
+import org.openstreetmap.atlas.geography.sharding.Sharding;
+import org.openstreetmap.atlas.tags.NaturalTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
 import org.openstreetmap.atlas.utilities.time.Time;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.MessageFormatter;
 
 /**
  * Main entry point to initiate raw {@link Atlas} country-slicing.
  *
  * @author mgostintsev
+ * @author samg
  */
 public class RawAtlasCountrySlicer
 {
+
+    // Logging constants
+    private static final String STARTED_TASK_MESSAGE = "Started {} for Shard {}";
+    private static final String COMPLETED_TASK_MESSAGE = "Finished {} for Shard {} in {}";
+    private static final String DYNAMIC_ATLAS_CREATION_TASK = "Dynamic Atlas Creation";
+
     private static final Logger logger = LoggerFactory.getLogger(RawAtlasCountrySlicer.class);
 
+    private final Sharding sharding;
+
+    private final Function<Shard, Optional<Atlas>> atlasFetcher;
+
     // The countries to be sliced with
     private final CountryBoundaryMap countryBoundaryMap;
 
     // The boundaries used for slicing
     private final Set<String> countries = new HashSet<>();
 
+    private final List<Shard> loadedShards = new ArrayList<>();
+
+    // Bring in all points -- this is important for proper sectioning later downstream, otherwise we
+    // will bring in geometry definitions with no underlying points
+    private static final Predicate<AtlasEntity> pointPredicate = entity -> entity instanceof Point;
+
+    // Bring in all relations that are tagged as Water or Coastline
+    private static final Predicate<AtlasEntity> relationPredicate = entity -> entity.getType()
+            .equals(ItemType.RELATION)
+            && Validators.isOfType(entity, NaturalTag.class, NaturalTag.WATER,
+                    NaturalTag.COASTLINE);
+
+    // Dynamic expansion filter will be a combination of the water relations and points
+    private static final Predicate<AtlasEntity> dynamicAtlasExpansionFilter = entity -> pointPredicate
+            .test(entity) || relationPredicate.test(entity);
+
     /**
-     * Slices against the given country and boundary map. Note: Will assume the entire world to be
-     * the target MultiPolygon to use.
+     * The default constructor for the old, pre-water relation pipeline-- this method will slice
```

Comment:
Would remove "pipeline" here..

Reply:
ah, was a clunky attempt to reference the entrypoints for relation slicing. will reword to be more specific.